src/nvim/ex_getln.c

```diff
@@ -1010,6 +943,11 @@ static uint8_t *command_line_enter(int firstc, long count, int indent, bool init
     msg_ext_clear_later();
   }
 
+  // Buffer redraw is needed if no command line
+  if (p_ch < 1 && !ui_has(kUICmdline) && !cmd_silent) {
+    redraw_all_later(NOT_VALID);
```

Comment:
Is `NOT_VALID` correct here? This will cause a full screen redraw when we only need to redraw the bottom 2 lines.